src/Properties.jl

```diff
@@ -0,0 +1,19 @@
+# special handling of message properties
+
+function Base.getproperty(msg::BasebandSignal, p::Symbol)
+  p === :signal && return transpose(reshape(getfield(msg, :signal), msg.channels, :))
+  getfield(msg, p)
+end
+
+function Base.setproperty!(msg::BasebandSignal, p::Symbol, v::AbstractVecOrMat)
+  if p === :signal
+    msg.channels = size(v, 2)
+    v = vec(transpose(v))
+    if eltype(v) == Complex{Float64}
+      v = ComplexF32.(v)
```

Comment:
```suggestion
      v = convert(Array{ComplexF32}, v)
```

Unlike the broadcast, this is a no-op if `v` is already of the requested type. 

```julia
julia> broadcast(a) = @time Float32.(a)
broadcast (generic function with 1 method)

julia> broadcast(Float32[1 2; 3 4]);
  0.000002 seconds (2 allocations: 96 bytes)

julia> conversion(a) = @time convert(Array{Float32}, a)
conversion (generic function with 1 method)

julia> conversion(Float32[1 2; 3 4]);
  0.000000 seconds
```

Reply:
Applied